Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,33 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
+    char days[64] = "", seconds[64] = "", microseconds[64] = "";
+    if (GET_TD_DAYS(self) != 0) {
+        sprintf(days, "days=%d", GET_TD_DAYS(self));
+    }
+
+    if (GET_TD_SECONDS(self) != 0 ||
+            (GET_TD_DAYS(self) == 0 && GET_TD_MICROSECONDS(self) == 0)) {
+        if (strlen(days) == 0) {
+            sprintf(seconds, "seconds=%d", GET_TD_SECONDS(self));
+        } else {
+            sprintf(seconds, ", seconds=%d", GET_TD_SECONDS(self));
+        }
```

Comment:
I've implemented a version using `PyList` and `PyUnicode`. I don't think I'm leaking any memory there, but since this is the first time I'm touching `PyObject`s, I would really appreciate a second opinion on it.